packages/state_queue/lib/src/state_queue.dart

```diff
@@ -76,15 +76,10 @@ abstract class StateQueue<T> extends ValueNotifier<T>
           await for (final nextState in event.updater(this.value)) {
             yield nextState;
           }
+
           // ignore: avoid_catches_without_on_clauses
-        } catch (e, stack) {
-          if (StateQueue.errorHandler != null) {
-            StateQueue.errorHandler(e, stack);
-          } else {
-            print(
-              'StateQueue: Error occured in `run` stream:\nimplementation: $this\n$e\n$stack',
-            );
-          }
+        } catch (error, stack) {
+          Zone.current.handleUncaughtError(error, stack);
```

Comment:
I think using `Zone` here might be a better general option here, as it allows for composition and is built into dart.

Reply:
Seems fine, especially since we just divert them to the same handler in the app anyway (which seems like the common case).

We just want to get them out of here, so that future `run`s still work.